carma_wm_ctrl/src/WMBroadcaster.cpp

```diff
@@ -463,17 +479,36 @@ cav_msgs::ControlRequest WMBroadcaster::controlRequestFromRoute(const cav_msgs::
 
   lanelet::GPSPoint gpsRoute = local_projector.reverse(localPoint); //If the appropriate library is included, the reverse() function can be used to convert from local xyz to lat/lon
 
-  cav_msgs::ControlRequest cR; /*Fill the latitude value in message cB with the value of lat */
-  cav_msgs::ControlBounds cB; /*Fill the longitude value in message cB with the value of lon*/
-  cB.latitude = gpsRoute.lat;
-  cB.longitude = gpsRoute.lon;
-
-//TODO: Update for new geofence spec
-  cB.offsets[0] = maxX - minX;
-  cB.offsets[1] = maxY - minY;
-
-  cR.bounds.push_back(cB);
-  ROS_WARN_STREAM("Got here safely");
+  cav_msgs::TrafficControlRequest cR; /*Fill the latitude value in message cB with the value of lat */
+  cav_msgs::TrafficControlBounds cB; /*Fill the longitude value in message cB with the value of lon*/
+  
+  cB.reflat = gpsRoute.lat;
+  cB.reflon = gpsRoute.lon;
+
+  cav_msgs::OffsetPoint offsetX;
+  offsetX.deltax = maxX - minX;
+  cav_msgs::OffsetPoint offsetY;
+  offsetY.deltay = maxY - minY;
+  cB.offsets[0] = offsetX;
+  cB.offsets[1] = offsetY;
+  cB.oldest =ros::Time::now();
```

Comment:
The semantic organization of OffsetPoint's changed. You want to create 3 offset points by setting there deltaX and deltaY to the following values
min, max
max, min
max, max


The min, min point is already handled as the lat/lon point.